azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/ISqlServerCreator.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+package com.microsoft.azure.toolkit.lib.sqlserver.service;
+
+import com.microsoft.azure.toolkit.lib.common.model.Region;
+import lombok.Getter;
+
+public interface ISqlServerCreator<T> {
+
+    ISqlServerCreator<T> withName(String name);
+
+    ISqlServerCreator<T> withResourceGroup(String resourceGroupName);
+
+    ISqlServerCreator<T> withRegion(Region region);
+
+    ISqlServerCreator<T> withAdministratorLogin(String administratorLogin);
+
+    ISqlServerCreator<T> withAdministratorLoginPassword(String administratorLoginPassword);
+
+    ISqlServerCreator<T> withEnableAccessFromAzureServices(boolean enableAccessFromAzureServices);
+
+    ISqlServerCreator<T> withEnableAccessFromLocalMachine(boolean enableAccessFromLocalMachine);
+
+    T commit();
```

Comment:
for commit, this is already an interface in common-lib

Reply:
updated.